db/migrate/20170307114233_add_collection_to_pages.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+class AddCollectionToPages < ActiveRecord::Migration
+  def change
+    add_reference :pages, :collection, index: true
+    add_foreign_key :pages, :collections
+
+    reversible do |change|
+      change.up do
+        Page::Landing.all.each do |landing_page|
+          key = landing_page.slug.split('/').last
+          key = key.blank? ? 'all' : key
+          collection = Collection.find_by_key(key)
```

Comment:
sure, I'll not do anything with the landing page in that case, however it'll not be a valid instance in accordance with the validations on the new landing page model. If there are landing pages like this we will most likely have to do a manual cleanup anyway.